### ISSUE TITLE
Style/FloatDivision is irun with `rubocop -a`

### ISSUE THREAD

**Issue by bigzed** [rubocop-hq/rubocop#9380]
## Expected behavior

I would suspect `rubocop -a` to be conservative and not break code by applying cops when ran in safe mode.

## Actual behavior

`rubocop -a` corrected

```ruby
a = '1.2'
b = '1.3'
c = '1.4'
d = a.to_f + (b.to_f / c.to_f)
#to
d = a.to_f + (b.to_f / c)
```
but when `c` is a `String` this raises `String can't be coerced into Float`.

## Steps to reproduce the problem

Reformat the above code with `rubocop -a`

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V   
1.8.1 (using Parser 3.0.0.0, rubocop-ast 1.4.0, running on ruby 2.5.8 x86_64-linux)
```
